src/nimble_kokkos.h

```diff
@@ -63,10 +63,7 @@ struct NimbleKokkosInitData {
 
 NimbleKokkosInitData NimbleKokkosInitializeAndGetInput(int argc, char* argv[]);
 
-void NimbleKokkosMain(std::shared_ptr<nimble_kokkos::MaterialFactory> material_factory,
```

Comment:
nimble_kokkos::MaterialFactory, nimble_kokkos::ContactInterface, nimble_kokkos::BlockMaterialInterfaceFactory, and nimble::Parser are all classes which are overriden in Extras; so some interface to create these polymorphic objects outside of NimbleSM and pass them in is needed